src/hooks/VoicePlayer/index.tsx

```diff
@@ -1,129 +1,168 @@
+  const [voicePlayerStore, voicePlayerDispatcher] = useReducer(voicePlayerReducer, voicePlayerInitialState);
+  const {
+    currentGroupKey,
+    currentPlayer,
+    audioStorage,
+  } = voicePlayerStore;
 
-  const addEventHandler = (event: VoicePlayerEvent): void => {
-    const { groupKey } = event;
-    setEventStorage((storage) => {
-      if (!storage?.[groupKey]) {
-        storage[groupKey] = [];
-      }
-      storage?.[groupKey].push(event);
-      return storage;
-    });
-  };
-  const removeEventHandler = (groupKey: string, handlerId: string): void => {
-    setEventStorage((storage) => {
-      if (!Array.isArray(storage?.[groupKey])) {
-        return storage;
-      }
-      return ({
-        ...storage,
-        [groupKey]: storage[groupKey].filter(({ id }) => id !== handlerId),
-      });
-    });
-  };
-  const triggerEvent = useCallback((eventType: VoicePlayerEventTypes, payload: VoicePlayerEventParams): void => {
-    const { groupKey } = payload;
-    if (Array.isArray(eventStorage?.[groupKey])) {
-      eventStorage[groupKey].map((playerEvent) => {
-        playerEvent?.[eventType]?.(payload);
-      });
+  const stop = (text = '') => {
+    if (currentGroupKey.includes(text)) {
+      pause(currentGroupKey);
     }
-  }, [eventStorage]);
-  const clearStates = (): void => {
-    setAudioPlayer(null);
-    setCurrentGroupKey('');
   };
 
-  const stop = useCallback((groupKey?: string): void => {
-    if (groupKey === undefined || (groupKey?.length > 0 && groupKey === currentGroupKey)) {
+  const pause = (groupKey: string) => {
+    if (currentGroupKey === groupKey && currentPlayer !== null) {
       currentPlayer?.pause();
-      clearStates();
     }
-  }, [currentPlayer, currentGroupKey]);
+  };
+
   const play = ({
-    audioFile,
-    playbackTime,
     groupKey,
+    audioFile = null,
+    audioFileUrl = '',
   }: VoicePlayerPlayProps): void => {
-    if (currentPlayer || currentGroupKey) {
-      stop();
+    if (groupKey !== currentGroupKey) {
+      pause(currentGroupKey);
     }
 
-    const audioPlayer = new Audio(URL?.createObjectURL?.(audioFile));
-    audioPlayer.currentTime = playbackTime;
-    audioPlayer.volume = 1;
-    audioPlayer.loop = false;
-    audioPlayer.onplaying = () => {
-      triggerEvent(VoicePlayerEventTypes.STARTED, {
-        groupKey,
-        playbackTime: audioPlayer?.currentTime,
-        duration: audioPlayer?.duration,
-      });
-    };
-    audioPlayer.onpause = () => {
-      triggerEvent(VoicePlayerEventTypes.STOPPED, {
-        groupKey,
-        playbackTime: audioPlayer?.currentTime,
-        duration: audioPlayer?.duration,
+    // Clear the previous AudioPlayer element
+    const voicePlayerRoot = document.getElementById(VOICE_PLAYER_ROOT_ID);
+    const voicePlayerAudioElement = document.getElementById(VOICE_PLAYER_AUDIO_ID);
+    if (voicePlayerAudioElement) {
+      voicePlayerRoot.removeChild(voicePlayerAudioElement);
+    }
+
+    new Promise((resolve) => {
```

Comment:
Here, I'm getting audioFile from three ways
1. already saved in the audioStorage
2. audioFile, the parameter of `play` (this is used in VoiceMessageInput)
3. audioFileUrl, the parameter of `play` (this is used in VoiceMessageItemBody)